src/main/java/org/mockito/internal/runners/DefaultInternalRunner.java

```diff
@@ -30,40 +30,50 @@ public DefaultInternalRunner(Class<?> testClass, final Supplier<MockitoTestListe
             private MockitoTestListener mockitoTestListener;
 
             protected Statement withBefores(FrameworkMethod method, Object target, Statement statement) {
-                // get new test listener and add add it to the framework
+                this.target = target;
+                // get new test listener and add it to the framework
                 mockitoTestListener = listenerSupplier.get();
                 Mockito.framework().addListener(mockitoTestListener);
-
                 // init annotated mocks before tests
                 MockitoAnnotations.initMocks(target);
-                this.target = target;
                 return super.withBefores(method, target, statement);
             }
 
             public void run(final RunNotifier notifier) {
                 RunListener listener = new RunListener() {
+                    private boolean started;
                     Throwable failure;
 
+                    @Override
+                    public void testStarted(Description description) throws Exception {
+                        started = true;
+                    }
+
+                    @Override
                     public void testFailure(Failure failure) throws Exception {
                         this.failure = failure.getException();
+                        // If the test fails during the setup, `testFinished` is never invoked
+                        // Therefore, if we have not started, cleanup the testlistener
+                        if (!started && mockitoTestListener != null) {
+                            Mockito.framework().removeListener(mockitoTestListener);
+                        }
                     }
 
+                    @Override
                     public void testFinished(Description description) throws Exception {
-                        super.testFinished(description);
```

Comment:
Because the class RunListener is documented that 
have no implementation.
It more like a interface with defaut empty implementation.

https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/runner/notification/RunListener.java